client-skeleton-common/src/main/java/eu/arrowhead/client/skeleton/common/ArrowheadService.java

```diff
@@ -0,0 +1,395 @@
+					logger.info("'{}' core service couldn't be retrieved due to the following reason: not registered by Serivce Registry", coreService.getServiceDefinition());
+					arrowheadContext.remove(coreService.getServiceDefinition() + ClientCommonConstants.CORE_SERVICE_DEFINITION_SUFFIX);
+					
+				} else {
+					final ServiceRegistryResponseDTO serviceRegistryResponseDTO = response.getBody().getServiceQueryData().get(0);
+					arrowheadContext.put(coreService.getServiceDefinition() + ClientCommonConstants.CORE_SERVICE_DEFINITION_SUFFIX, new CoreServiceUri(serviceRegistryResponseDTO.getProvider().getAddress(),
+							serviceRegistryResponseDTO.getProvider().getPort(), serviceRegistryResponseDTO.getServiceUri()));					
+				}
+				
+			} catch (final  ArrowheadException ex) {
+				logger.info("'{}' core service couldn't be retrieved due to the following reason: {}", coreService.getServiceDefinition(), ex.getMessage());
+			}			
+		}
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Sends a http(s) request to the 'echo' end point of the given Arrowhead Core System.
+	 * 
+	 * @param coreSystem CoreSystem enum which represents an Arrowhead Core System
+	 * @return the response received from core system server or null when the specified core system has no public service or it is not known by Service Registry Core System
+	 * @throws ArrowheadException when internal server error happened at Service Registry or at the given core system
+	 * @throws UnavailableServerException when Service Registry Core System or the given core system is not available
+	 */
+	public ResponseEntity<String> echoCoreSystem(final CoreSystem coreSystem) {		
+		String address = null;
+		Integer port = null;
+		String coreUri = null;
+		
+		if (coreSystem == CoreSystem.SERVICE_REGISTRY) {
+			address = serviceReqistryAddress;
+			port = serviceRegistryPort;
+			coreUri = CommonConstants.SERVICE_REGISTRY_URI;
+			
+		} else {			
+			final List<CoreSystemService> publicServices = getPublicServicesOfCoreSystem(coreSystem);			
+			if (publicServices.isEmpty()) {
+				logger.debug("'{}' core system has no public service.", coreSystem.name());
+				return null;
+				
+			} else {				
+				final ResponseEntity<ServiceQueryResultDTO> srResponse = queryServiceReqistryByCoreService(publicServices.get(0));
+				
+				if (srResponse.getBody().getServiceQueryData().isEmpty()) {
+					logger.debug("'{}' core system not known by Service Registry", coreSystem.name());
+					return null;
+				} else {
+					address = srResponse.getBody().getServiceQueryData().get(0).getProvider().getAddress();
+					port = srResponse.getBody().getServiceQueryData().get(0).getProvider().getPort();
+					coreUri = publicServices.get(0).getServiceUri().split("/")[1];
+				}				
+			}			
+		}
+		
+		return httpService.sendRequest(Utilities.createURI(getUriScheme(), address, port, coreUri + CommonConstants.ECHO_URI), HttpMethod.GET, String.class);
```

Comment:
in case of boolean return value, you should catch exceptions and log it, then return with false in the catch block